### ISSUE TITLE
Bug when using stack's docker integration (command_ hangs)

### ISSUE THREAD

**Issue by SandiCat** [ndmitchell/shake#735]
I'm building a stack project using shake. Stack is configured to use docker ([see here](https://docs.haskellstack.org/en/stable/docker_integration/)). Building (`stack build`) works at the shell but not through shake. Shake gets stuck at the first stack command it tries to run. The command completes fully, i.e. all the expected output is displayed, but then shake gets stuck and doesn't run any further actions.

Here is a minimal reproduction:
1. clone https://github.com/SandiDusic/shake-stack-docker-bug-repro
1. **edit:** optionally, uncomment the line in `my-app/stack.yaml` that changes the docker image. I can't try it out right now, but I think the problem persists with any image, so there's no need to download the default giant `fpco/stack-build` images.
1. in the `build-system` directory, build the build system with `stack build`
1. run the build system with `stack exec build -- build`
1. the project (`my-app`) builds completely, but then shake hangs (`putQuiet "done building"` never gets run)
1. try running `stack build` directly in `my-app/` - it works (though NB it might show no output if it doesn't have to recompile, at least it doesn't for me)
1. try removing `docker: enable: true` from `my-app/stack.yaml`. the build system will work.

Info about my system:
```
Docker version 19.03.5-ce, build 633a0ea838
Linux pc 4.19.88-1-MANJARO #1 SMP PREEMPT Thu Dec 5 11:04:44 UTC 2019 x86_64 GNU/Linux
Version 2.1.3, Git revision 636e3a759d51127df2b62f90772def126cdf6d1f (7735 commits) x86_64 hpack-0.31.2
```

I am having some other troubles with stack's docker integration (https://github.com/commercialhaskell/stack/issues/5118) so my setup might be defective somehow.

**Comment by ndmitchell** [comment]
I don't have a Linux environment so will struggle to reproduce this. I suggest running shake with `-VVV` which will print lots of extra information - in particular it might give you a clue what step it freezes on. If you upload that trace it should give us a hint. If you change the command to run anything other than stack does it work? If so, that sounds like Stack is defective (if not, worth investigating further). Perhaps add more verbosity arguments to stack itself too?

**Comment by SandiDusic** [comment]
> If you change the command to run anything other than stack does it work?

Well, stack itself works, if it's not running a docker. `echo` works as well. So does docker, for example replacing the `command_` with:
```command_ [] "docker" ["run", "--rm", "alpine", "ls", "/bin"]```

>  I suggest running shake with -VVV which will print lots of extra information - in particular it might give you a clue what step it freezes on.

Here's the whole output with max verbosity on both shake and stack:
https://gist.github.com/SandiDusic/01bf60a1cc1044656b5035f03a790b98

Notice that I terminated manually at the end, the command hanged before that. I also updated the repo, adding the `-v` option to stack so this output can be seen. 

> If so, that sounds like Stack is defective

I wouldn't know how to open a useful issue with them, aside from linking this one. Any idea what could make a program misbehave when run from shake? Stack does work fine when running the command manually.
